hips/hip-0006.md

```diff
@@ -0,0 +1,253 @@
+
+Here is an example of a registry config file using a credential helper:
+
+```
+{
+	"auths": {
+		"example.com": {}
+	},
+	"credsStore": "osxkeychain"
+}
+```
+
+Here is an example of a registry config file without a credential helper:
+
+```
+{
+	"auths": {
+		"example.com": {
+			"auth": "<password>",
+			"email": "<username>"
+		}
+	}
+}
+```
+
+So, while a new attack vector has been introduced, it should not be considered any less safe than using `docker login`.
+In fact, he underlying code is leveraging docker/cli, and existing logins created with `docker login` are leveraged if not found in Helm's config.
+
+Note: this is already implemented in the current OCI feature set.
+
+## How to teach this
+
+Following the implementation of all facets of this HIP, extensive documentation should be added to the Helm website on how to leverage OCI.
+
+Additionally, sites containing community charts, such as [Artifact Hub](https://artifacthub.io/), should enable providers to distribute charts over OCI, and display friendly copy-paste instructions for how to download an OCI-based chart.
+
+Helm and charts maintainers should be educated on how to make a transition from classic chart repositories to registries.
+In short the workflow for uploading a chart goes from `helm package` (+ custom upload) to `helm package && helm registry login && helm push`.
+
+## Reference implementation
+
+The features are currently under development. 
+
+Here are the following outstanding tasks:
+
+- We need to implement helm push, which does not exist
+- Helm needs to implement OCI Getter support so that helm dependency and helm pull can support OCI registries (see https://github.com/helm/helm/pull/8843)
+- Support for provenance files must be added
+- `helm chart save`, `helm chart export`, `helm chart list`, `helm chart remove`, `helm chart pull`, and `helm chart push` need to be removed
+- The Helm documentation for OCI support needs to be updated (https://helm.sh/docs/topics/registries/))
+
+## Rejected ideas
+
+### Will I be able to use custom tags such as "latest"?
+
+The implications of this are not yet fully understood. For now, it will be strictly enforced that the tag on the reference will match the version found in `Chart.yaml`. This functionality may be added at a later time.
+
+### Will I be able to pull a chart using the `:<tag>` syntax?
+
+It's unclear if users will be able to pull charts with a specific version using the `:<tag>` syntax (vs. `--version <version>`). This seems relatively harmless to allow both, but this may have implications on dependencies and other areas.
+
+## Open issues
+
+The issues below are still unresolved.
+
+### What is the correct media type to use for the chart content layer?
+
+The media type for the chart content layer is currently `application/tar+gzip`, however it has been pointed out that [this is not an official type in the IANA database](https://www.iana.org/assignments/media-types/media-types.xhtml).
```

Comment:
Looks like this was addressed in an earlier comment and can be resolved by using option A.